session/session_test.go

```diff
@@ -4305,23 +4320,41 @@ func (s *testTxnStateSuite) TestBlocked(c *C) {
 		tk2.MustExec("select * from t where a = 1 for update;")
 		tk2.MustExec("commit;")
 	}()
-	time.Sleep(200 * time.Millisecond)
+	time.Sleep(100 * time.Millisecond)
 	c.Assert(tk2.Se.TxnInfo().State, Equals, txninfo.TxnLockWaiting)
 	c.Assert(tk2.Se.TxnInfo().BlockStartTime, NotNil)
 	tk.MustExec("commit;")
 }
 
-func (s *testTxnStateSuite) TestLenAndSize(c *C) {
+func (s *testTxnStateSuite) TestCommitting(c *C) {
 	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk2 := testkit.NewTestKitWithInit(c, s.store)
 	tk.MustExec("create table t(a int);")
+	tk.MustExec("insert into t(a) values (1), (2);")
 	tk.MustExec("begin pessimistic;")
-	tk.MustExec("insert into t(a) values (1);")
-	info := tk.Se.TxnInfo()
-	c.Assert(info.EntriesCount, Equals, int64(1))
-	c.Assert(info.EntriesSize, Equals, int64(29))
-	tk.MustExec("insert into t(a) values (2);")
-	info = tk.Se.TxnInfo()
-	c.Assert(info.EntriesCount, Equals, int64(2))
-	c.Assert(info.EntriesSize, Equals, int64(58))
+	tk.MustExec("select * from t where a = 1 for update;")
+	go func() {
+		tk2.MustExec("begin pessimistic")
+		c.Assert(tk2.Se.TxnInfo(), NotNil)
+		tk2.MustExec("select * from t where a = 2 for update;")
+		failpoint.Enable("github.com/pingcap/tidb/session/mockSlowCommit", "sleep(200)")
```

Comment:
For failpoint tests, please disable the failpoint after the test, and put the test in a serial suite to avoid affecting concurrent cases.